CRM/Mailing/BAO/Mailing.php

```diff
@@ -2437,6 +2437,33 @@ public static function del($id) {
    * @throws CRM_Core_Exception
    */
   public static function self_hook_civicrm_pre(\Civi\Core\Event\PreEvent $event) {
+    if ($event->action === 'create') {
+      $params = &$event->params;
+      $params['created_id'] ??= CRM_Core_Session::singleton()->getLoggedInContactID();
+      $params['override_verp'] ??= !Civi::settings()->get('track_civimail_replies');
+      $params['visibility'] ??= 'Public Pages';
+      $params['dedupe_email'] ??= Civi::settings()->get('dedupe_email_default');
+
+      $params['forward_replies'] ??= FALSE;
+      $params['auto_responder'] ??= FALSE;
```

Comment:
These two shouldn't be needed since defaults were added to the schema:
https://github.com/civicrm/civicrm-core/blob/30d49c2734cf72d6b0a920e494ac165b7dcee0cb/xml/schema/Mailing/Mailing.xml#L264-L278

Reply:
does the schema handle created_id (it should but I have a feeling we couldn't implement that on an earlier mysql version)

Reply:
@eileenmcnaughton I'm not aware of any entity that has the capability to auto-insert current user as a sql default. I belive it's always done by BAO logic. What @seamuslee001 did here looks consistent with other entitie, e.g.
 
https://github.com/civicrm/civicrm-core/blob/92a32996710ed8f19ac9ff30e38c566a05bd9ac2/CRM/Contact/BAO/SavedSearch.php#L252-L255

Reply:
OK